TypeScript.py

```diff
@@ -1501,6 +1531,92 @@ def run(self, text):
             formatRange(text, view, rblineStart, ralineEnd)
 
 
+class NavToCommand(sublime_plugin.WindowCommand):
+    navto_panel_started = False
+
+    # indicate if the insert_text command has finished pasting text into the textbox.
+    # during which time the on_modified callback shouldn't do angthing
+    insert_text_finished = False
+
+    # when a new search string comes in, the command state needs to be reset and then
+    # restart. However the reset cannot be reset after the "run" method. Because
+    # the exeuation order of the "run" and "on_done" methods are not fixed, this flag is
+    # set to make sure the state reset to be done before run
+    reset_already = False
+    input_text = ""
+
+    @classmethod
+    def reset(cls):
+        cls.navto_panel_started = False
+        cls.insert_text_finished = False
+
+    def is_enabled(self):
+        return is_typescript(self.window.active_view())
+
+    def run(self, input_text = ""):
+        logger.log.debug("start running navto wiht text: %s" % input_text)
+
+        NavToCommand.reset_already = False
+        
+        self.window.run_command("hide_overlay")
+
+        if NavToCommand.reset_already:
+            NavToCommand.reset_already = False
+        else:
+            logger.log.debug("reset in run")
+            NavToCommand.reset()
+            NavToCommand.reset_already = True
+
+        NavToCommand.input_text = input_text        
+        NavToCommand.navto_panel_started = True
+
+        # query text is not always equal to input text. When no input_text is given
+        # alternative query text is used to ensure the panel stay active
```

Comment:
I think this is misleading unless I'm misunderstanding the logic. I think you mean to say that we still want to show results even if there is no text

Does the navto search bar actually lose focus or disappear if there is no text?